questionpy_sdk/webserver/routes/base.py

```diff
@@ -0,0 +1,31 @@
+#  This file is part of the QuestionPy SDK. (https://questionpy.org)
+#  The QuestionPy SDK is free software released under terms of the MIT license. See LICENSE.md.
+#  (c) Technische Universität Berlin, innoCampus <info@isis.tu-berlin.de>
+
+from typing import TYPE_CHECKING, Any, Generic, TypeVar
+
+from aiohttp import web
+from pydantic import BaseModel
+
+from questionpy_sdk.webserver.constants import REQUEST_CONTROLLER_KEY
+
+if TYPE_CHECKING:
+    from questionpy_sdk.webserver.controllers.base import BaseController
+
+CT = TypeVar("CT", bound="BaseController")
+
+
+class BaseView(web.View, Generic[CT]):
```

Comment:
Seit Python 3.12 haben Generics eine eigene Syntax:

```suggestion
class BaseView[CT](web.View):
```

Die Zukunft ist jetzt

Reply:
Die Zukunft lässt im SDK noch auf sich warten, wir müssen mal die aktuelle `ruff_defaults.toml` aus qpy-dev synchronisieren (und ganz nebenbei auch dringend mal ruff updaten).

```toml
# ruff_defaults.toml
target-version = "py311"
```

Die Python-Versionen sind irgendwie eh gerade außer Rand und Band. Server will 3.11, SDK 3.12. Ruff ist im SDK noch auf 3.11, GitHub Actions waren noch auf 3.11-13. (Hatte ich in meinem [letzten Commit](https://github.com/questionpy-org/questionpy-sdk/commit/eea89967913c3953e0b36e89a44bb13791dc2446#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR16) angeglichen zum SDK)

Wollten wir nicht mal die letzten 3 Python-Versionen supporten oder so ähnlich? Ich bin verwirrt.

Reply:
Hm, ja, ich habe bei https://github.com/questionpy-org/questionpy-server/pull/145 ehrlicherweise nicht so ganz an Debian gedacht, aber tatsächlich enthält Debian Bookworm nur 3.11, ich glaube @MartinGauk hatte mal vorgeschlagen, davon wir uns daran orientieren sollten. Wahrscheinlich kommt Debian Trixie noch vor dem nächsten Semester raus und enthält 3.13, aber ich kann den Bump auch zurückrollen, was meint ihr?

Sicherlich macht es aber Sinn, das außerhalb dieses PRs zu betrachten ^^